### ISSUE TITLE
RNPM link android break

### ISSUE THREAD

**Issue by DeDuckProject** [Microsoft/react-native-code-push#571]
- react-native-code-push version: ^1.15.0-beta
- react-native version: 0.30.0
- Does this reproduce on a debug build or release build? both
- Does this reproduce on a simulator, or only on a physical device? all
- Repro steps and/or stack trace:

run the react-native link command

MainApplication in android looks like this:

``` java
public class MainApplication extends Application implements ReactApplication {

  public static final int PAY_PAL_REQUEST_ID = 9;
  public PayPalPackage payPalPackage;

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }

    @Override
    protected boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      payPalPackage = new PayPalPackage(PAY_PAL_REQUEST_ID);
      return Arrays.<ReactPackage>asList(
              new MainReactPackage(),
            new CodePush(this.getResources().getString(R.strings.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG),
              new ReactMaterialKitPackage(),
              new ReactNativePushNotificationPackage(),
              payPalPackage
      );
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }
}

```

`new CodePush(this.getResources().getString(R.strings.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG),` brings an error as "this" refers to ReactNativeHost and not MainApplication.
I've inserted the key manually.

**Comment by silhouettes** [comment]
Thanks for filing this, these are extremely helpful! Will have a look at it over the next few days.

**Comment by buptkang** [comment]
@DeDuckProject it seems there are two bugs in the react native cli that we fixed in 0.33 (https://github.com/facebook/react-native/commit/f31b6ba953b3caa4f964537b9badef9f4617c530 and https://github.com/facebook/react-native/commit/1168d0db45dc6105a4e88a7749ad5867d37049d2). If you still need to use your current version (0.30.0), you should make the two changes to replace the below code:

```
new CodePush(this.getResources().getString(R.strings.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG)
```

as

```
new CodePush(getResources().getString(R.string.reactNativeCodePush_androidDeploymentKey), getApplicationContext(), BuildConfig.DEBUG)
```

Otherwise, you can upgrade the React-Native version to above 0.33, which will not raise this issue again. We've also updated the README so that other people don't run into the same issue - thanks so much for filing this!

**Comment by DeDuckProject** [comment]
Thanks
